From 08aae283a2c87c054f42bc441963d0739208087b Mon Sep 17 00:00:00 2001
From: Ross Wille <wille@freescale.com>
Date: Fri, 5 Dec 2008 16:14:29 -0700
Subject: [PATCH] ENGR00101071 MX25: Do not assume UART4 pins are GPIO

The 4 pins used by UART4 (ttymxc/3) were being set to GPIO
when the UART was deactivated, but since these pins cannot
be configured as GPIO, this generated "invalid GPIO" errors.
This commit fixes this problem by returning these pins to
their default functions.

Signed-off-by: Quinn Jensen <quinn.jensen@freescale.com>
---
 arch/arm/mach-mx25/mx25_3stack_gpio.c |   13 ++++---------
 1 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-mx25/mx25_3stack_gpio.c b/arch/arm/mach-mx25/mx25_3stack_gpio.c
index 9b8afdf..cb72e90 100644
--- a/arch/arm/mach-mx25/mx25_3stack_gpio.c
+++ b/arch/arm/mach-mx25/mx25_3stack_gpio.c
@@ -170,15 +170,10 @@ void gpio_uart_inactive(int port, int no_irda)
 		mxc_free_iomux(MX25_PIN_KPP_ROW3, MUX_CONFIG_GPIO);
 		break;
 	case 3:
-		mxc_request_gpio(MX25_PIN_LD8);
-		mxc_request_gpio(MX25_PIN_LD9);
-		mxc_request_gpio(MX25_PIN_LD10);
-		mxc_request_gpio(MX25_PIN_LD11);
-
-		mxc_free_iomux(MX25_PIN_LD8, MUX_CONFIG_GPIO);
-		mxc_free_iomux(MX25_PIN_LD9, MUX_CONFIG_GPIO);
-		mxc_free_iomux(MX25_PIN_LD10, MUX_CONFIG_GPIO);
-		mxc_free_iomux(MX25_PIN_LD11, MUX_CONFIG_GPIO);
+		mxc_free_iomux(MX25_PIN_LD8, MUX_CONFIG_FUNC);
+		mxc_free_iomux(MX25_PIN_LD9, MUX_CONFIG_FUNC);
+		mxc_free_iomux(MX25_PIN_LD10, MUX_CONFIG_FUNC);
+		mxc_free_iomux(MX25_PIN_LD11, MUX_CONFIG_FUNC);
 		break;
 	case 4:
 		mxc_request_gpio(MX25_PIN_CSI_D2);
-- 
1.5.4.4

